### PR TITLE
Bazel updates for Harmonic

### PR DESCRIPTION
**Pull request by mjcarroll**

### CODE REVIEW

File: tools/gz_msgs_generate_factory.py [hunk 1]
```diff
@@ -131,45 +140,55 @@ def main(argv=sys.argv[1:]):
 
     args = parser.parse_args(argv)
 
-    headers = []
-    registrations = []
+    package_re = re.compile('^package (.*);$')
+    message_re = re.compile('message (.*)')
```
Comment (shameekganguly) [inline tools/gz_msgs_generate_factory.py:144]: Maybe use message_re = re.compile(r'message (\w*)\s?{?$') so we handle message classes linted with "{" on the same line

**Review by caguero — COMMENTED** [review]
@mjcarroll , how can we test it?

**Comment by mjcarroll** [comment]
@caguero CI running here: https://github.com/gazebosim/gz-bazel/pull/75

File: BUILD.bazel [hunk 2]
```diff
@@ -138,3 +158,100 @@ test_sources = glob(
...
+    ],
+    package = "msgs",
+)
+
+gz_configure_file(
+    name = "msgs_rb",
+    src = "core/src/cmd/cmdmsgs.rb.in",
+    out = "cmdmsgs.rb",
+    defines = [
+        "library_location=libgz-msgs.so",
```
Comment (shameekganguly) [inline BUILD.bazel:239]: Do we need to add PROJECT_VERSION_FULL=10.1.1?

File: core/generator_lite/Generator.hh [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+#include <google/protobuf/compiler/code_generator.h>
+#include <string>
+
+namespace google {
+namespace protobuf {
+namespace compiler {
+namespace cpp {
+/// \cond
+/// \brief Google protobuf message generator for igntion::msgs
```
Comment (shameekganguly) [inline core/generator_lite/Generator.hh:29]: ignition::msgs -> gz::msgs

File: core/generator_lite/Generator.hh [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  /// \param[in] _parameter Unused string value
+  /// \param[in] _generatorContext Output directory.
+  /// \param[in] _error Unused string value
+  /// \return true if successful, otherwise sets _error and returns false
+  public: virtual bool Generate(const FileDescriptor *_file,
+              const std::string &_parameter,
+              OutputDirectory *_generatorContext,
+              std::string *_error) const;
+
+  // private: GOOGLE_DISALLOW_EVIL_CONSTRUCTORS(Generator);
```
Comment (shameekganguly) [inline core/generator_lite/Generator.hh:50]: remove comment

File: core/src/impl/InstallationDirectories.cc [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+#include <string>
+
+#include <gz/msgs/InstallationDirectories.hh>
+
+namespace gz::msgs
+{
+inline namespace GZ_MSGS_VERSION_NAMESPACE {
+std::string getInstallPrefix() { return {};}
+}  // namespcae
```
Comment (shameekganguly) [inline core/src/impl/InstallationDirectories.cc:26]: nit: namespace typo

**Review by shameekganguly — APPROVED** [review]

File: BUILD.bazel [hunk 6]
```diff
@@ -91,26 +90,47 @@ proto_library(
...
 )
 
-gz_msgs_generate(
+gz_proto_library(
     name = "gzmsgs_cc_proto",
+    proto_deps = [":gzmsgs_proto"],
     deps = [
-        ":gzmsgs_proto",
-        "@com_google_protobuf//:any_proto",
+        "@com_google_protobuf//:protobuf",
```
[the remark this answers is not in the document]
Comment (mjcarroll) [reply BUILD.bazel:97]: I believe it's not a target in this version of protobuf:

```
$ bazel query "@com_google_protobuf//:*"
@com_google_protobuf//:BUILD
@com_google_protobuf//:any_proto
@com_google_protobuf//:api_proto
@com_google_protobuf//:cc_test_protos_genproto
@com_google_protobuf//:cc_toolchain
@com_google_protobuf//:cc_wkt_protos
@com_google_protobuf//:cc_wkt_protos_genproto
@com_google_protobuf//:compiler_plugin_proto
@com_google_protobuf//:descriptor_proto
@com_google_protobuf//:duration_proto
@com_google_protobuf//:empty_proto
@com_google_protobuf//:field_mask_proto
@com_google_protobuf//:java_toolchain
@com_google_protobuf//:javalite_toolchain
@com_google_protobuf//:linux-aarch64
@com_google_protobuf//:linux-ppc
@com_google_protobuf//:linux-s390x
@com_google_protobuf//:linux-x86_32
@com_google_protobuf//:linux-x86_64
@com_google_protobuf//:macos-aarch64
@com_google_protobuf//:macos-x86_64
@com_google_protobuf//:objectivec
@com_google_protobuf//:protobuf
@com_google_protobuf//:protobuf_headers
@com_google_protobuf//:protobuf_java
@com_google_protobuf//:protobuf_java_util
@com_google_protobuf//:protobuf_javalite
@com_google_protobuf//:protobuf_lite
@com_google_protobuf//:protobuf_objc
@com_google_protobuf//:protobuf_python
@com_google_protobuf//:protobuf_python_genproto
@com_google_protobuf//:protoc
@com_google_protobuf//:protoc_lib
@com_google_protobuf//:python_common_test_protos_genproto
@com_google_protobuf//:python_specific_test_protos_genproto
@com_google_protobuf//:source_context_proto
@com_google_protobuf//:struct_proto
@com_google_protobuf//:timestamp_proto
@com_google_protobuf//:type_proto
@com_google_protobuf//:well_known_type_protos
@com_google_protobuf//:windows-x86_32
@com_google_protobuf//:windows-x86_64
@com_google_protobuf//:wrappers_proto
```

File: BUILD.bazel [hunk 7]
```diff
@@ -3,28 +3,41 @@ load(
...
+licenses(["notice"])
 
 exports_files(["LICENSE"])
 
 gz_configure_header(
     name = "msgs_config_hh",
-    src = "include/gz/msgs/config.hh.in",
+    src = "core/include/gz/msgs/config.hh.in",
+    out = "include/gz/msgs/config.hh",
     cmakelists = ["CMakeLists.txt"],
```
[the remark this answers is not in the document]
Comment (mjcarroll) [reply BUILD.bazel:41]: In the bazel implementation, this is being read from the workspace and inserted via `gz_configure_header`: https://github.com/gazebosim/gz-bazel/blob/da8334877062ad1fdf3bca41c5ce40af28207d3a/skylark/gz_configure_file.py#L141-L154

File: BUILD.bazel [hunk 8]
```diff
@@ -138,3 +158,100 @@ test_sources = glob(
...
+        "@gtest//:gtest_main",
+    ],
+)
+
+gz_configure_file(
+    name = "msgs_yaml",
+    src = "conf/msgs.yaml.in",
+    out = "msgs.yaml",
+    defines = [
+        "PROJECT_NAME_NO_VERSION_LOWER=msgs",
```
[the remark this answers is not in the document]
Comment (mjcarroll) [reply BUILD.bazel:228]: Same as above

File: core/generator_lite/Generator.hh (hunk 3 above)
Comment (mjcarroll) [reply core/generator_lite/Generator.hh:29]: Done

File: core/generator_lite/Generator.hh (hunk 4 above)
Comment (mjcarroll) [reply core/generator_lite/Generator.hh:50]: Done

File: core/src/impl/InstallationDirectories.cc (hunk 5 above)
Comment (mjcarroll) [reply core/src/impl/InstallationDirectories.cc:26]: Done

**Pull request merged**